### ISSUE TITLE
"0 delta" daylight periods in TimeZoneInfo are annoying

### ISSUE THREAD

**Issue by jskeet** [nodatime/nodatime#746]
Consider these BCL adjustment rules for America/Louisville, as seen on Linux under Mono:

```text
[1973-01-01 - 1973-1231]; Delta: 01:00:00
  DST start: Month: 4; Week: 1; Day: 29; DoW: Sunday; Time: 02:00; Fixed? True
  DST end: Month: 10; Week: 1; Day: 28; DoW: Sunday; Time: 02:00; Fixed? True
[1974-01-01 - 1974-1231]; Delta: 00:00:00
  DST start: Month: 1; Week: 1; Day: 6; DoW: Sunday; Time: 02:00; Fixed? True
  DST end: Month: 10; Week: 1; Day: 27; DoW: Sunday; Time: 02:00; Fixed? True
[1975-01-01 - 1975-1231]; Delta: 01:00:00
  DST start: Month: 2; Week: 1; Day: 23; DoW: Sunday; Time: 02:00; Fixed? True
  DST end: Month: 10; Week: 1; Day: 26; DoW: Sunday; Time: 02:00; Fixed? True
```

Note the "Delta: 00:00:00". That means "use the standard offset for the time zone, but let's pretend it's DST." Noda Time will get the right offsets, but end up saying that it's not on DST. I believe this is due to TimeZoneInfo not having a (public) representation of "standard time changes by rule".

Possible solution: if there's a delta of 0, pretend that the standard offset (for this ZoneInterval) is 1 hour off, and that there's a 1 hour delta.

**Comment by jskeet** [comment]
Looking, this may be a problem for America/Resolute as well. Basically, our IsDaylight property perhaps incorrectly uses "0 savings" as an indication of DST. Needs more research - perhaps `ZoneInterval` needs an explicit `bool` in it. (And then we need to think about the nzd format. Sigh.)

**Comment by jskeet** [comment]
@mj1856 @malcolmr: Any thoughts on this?

I think we either have to say "IsDaylight isn't the same as Savings==0" and keep a separate bool in ZoneInterval, or we have to effectively invent data within `BclDateTimeZone`, e.g. if we have time zone standard = +3 and we see a savings of 0, we say that's equivalent to a standard of +2 and a savings of +1. I've a horrible feeling we couldn't do that within the ZoneRecurrence, as we need both recurrences for a StandardDaylightAlternatingMap to have the same standard offset, but we could fake it up somewhere.

Ironically, I've a sneaking suspicion that that's exactly what the `TimeZoneInfo` adjustment rules are *trying* to map - here's TZDB's rules for the same period:

```text
-5:00	US	E%sT	1974 Jan  6  2:00
-6:00	1:00	CDT	1974 Oct 27  2:00
-5:00	US	E%sT
```

**Comment by jskeet** [comment]
Fixed by #750.
